internal/services/azurestackhci/stack_hci_extension_resource_test.go

```diff
@@ -0,0 +1,219 @@
+			),
+		},
+		data.ImportStep("protected_settings"),
+	})
+}
+
+func TestAccStackHCIExtension_complete(t *testing.T) {
+	arcSettingId := os.Getenv(arcSettingIdEnv)
+	if arcSettingId == "" {
+		t.Skipf("skipping since %q has not been set", arcSettingIdEnv)
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_stack_hci_extension", "test")
+	r := StackHCIExtensionResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("protected_settings"),
+	})
+}
+
+func TestAccStackHCIExtension_update(t *testing.T) {
+	arcSettingId := os.Getenv(arcSettingIdEnv)
+	if arcSettingId == "" {
+		t.Skipf("skipping since %q has not been set", arcSettingIdEnv)
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_stack_hci_extension", "test")
+	r := StackHCIExtensionResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("protected_settings"),
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("protected_settings"),
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("protected_settings"),
+	})
+}
+
+func TestAccStackHCIExtension_requiresImport(t *testing.T) {
+	arcSettingId := os.Getenv(arcSettingIdEnv)
+	if arcSettingId == "" {
+		t.Skipf("skipping since %q has not been set", arcSettingIdEnv)
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_stack_hci_extension", "test")
+	r := StackHCIExtensionResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (r StackHCIExtensionResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	clusterClient := client.AzureStackHCI.Extensions
+	id, err := extensions.ParseExtensionID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clusterClient.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	return pointer.To(resp.Model != nil), nil
+}
+
+func (r StackHCIExtensionResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_stack_hci_extension" "test" {
+  name           = "acctest-shce-%[2]s"
+  arc_setting_id = %[3]q
+  publisher      = "Microsoft.EnterpriseCloud.Monitoring"
+  type           = "MicrosoftMonitoringAgent"
+
+  protected_settings = <<PROTECTED_SETTINGS
```

Comment:
`basic` should not contain optional properties: `protected_settings` and `settings`

Reply:
fixed